packages/core/src/plugins/cleanOutput.ts

```diff
@@ -43,7 +44,26 @@ export const pluginCleanOutput = (): RsbuildPlugin => ({
       }
     };
 
-    api.onBeforeBuild(clean);
-    api.onBeforeStartDevServer(clean);
+    const cleanAll = async () => {
+      const distPaths = Object.entries(api.context.environments).reduce<
+        Array<{
+          environmentName: string;
+          distPath: string;
+        }>
+      >((total, [environmentName, curr]) => {
+        if (!total.find((t) => t.distPath === curr.distPath)) {
+          total.push({
+            environmentName,
+            distPath: curr.distPath,
+          });
+        }
+        return total;
+      }, []);
+
+      await Promise.all(distPaths.map((d) => clean(d.environmentName)));
```

Comment:
Can we just pass the `distPath`?

```suggestion
      await Promise.all(distPaths.map((distPath) => clean(distPath)));
```

Reply:
environmentName  is also used to read environemt `output.cleanDistPath`  config.